// md-editor-v3/lib/es/NormalToolbar.mjs
import { defineComponent as r, createVNode as l } from "vue";
import { p as a } from "./chunks/config.mjs";
import { g as c } from "./chunks/vue-tsx.mjs";
const m = {
  title: {
    type: String,
    default: ""
  },
  // 展示在工具栏的内容，通常是个图标
  trigger: {
    type: [String, Object]
  },
  onClick: {
    type: Function
  },
  /**
   * ==没有意义，仅用于规避克隆组件自动嵌入insert方法时，传入的是该组件而产生的waring
   */
  insert: {
    type: Function
  },
  language: {
    type: String
  },
  theme: {
    type: String
  }
  /**
   * ==结束
   */
}, e = /* @__PURE__ */ r({
  name: "NormalToolbar",
  props: m,
  emits: ["onClick"],
  setup(t, n) {
    return () => {
      const o = c({
        props: t,
        ctx: n
      }, "trigger");
      return l("div", {
        class: `${a}-toolbar-item`,
        title: t.title,
        onClick: (i) => {
          t.onClick instanceof Function ? t.onClick(i) : n.emit("onClick", i);
        }
      }, [o]);
    };
  }
});
e.install = (t) => (t.component(e.name, e), t);
export {
  e as default
};
